### ISSUE TITLE
Emacs rendering bug can cause corruption in display of usernames in margins

### ISSUE THREAD

**Issue by mekeor** [alphapapa/ement.el#48]
Sometimes, for example when scrolling around within a room-buffer, nicknames are rendered incorrectly, i.e. they overlap. See screenshot below.

![2021-12-30-081326_1920x1080_scrot](https://user-images.githubusercontent.com/60295641/147729862-752ba6d1-d382-47f0-aa52-adca31952aa8.png)

**Comment by Gavinok** [comment]
Having the same issue here.

**Comment by alphapapa** [comment]
This is a margin-rendering bug in some versions of Emacs.  I don't know of any workarounds, other than configuring Ement to not use the margin for usernames.  (Maybe using a different GUI toolkit in Emacs would help, but I haven't tested that.)

If you can, please check the Emacs bug tracker for relevant bugs, or file a new one if necessary, and link it here.  Thanks.

**Comment by mekeor** [comment]
Do I experience this because I'm using Emacs with the Athena/Lucid graphical toolkit (`--with-x-toolkit=athena`)? @Gavinok, do you use Emacs-Lucid/Athena, too?

@alphapapa, how can I configure "Ement to not use the margin for usernames"? Which config-variable are you refering to?

When I set `ement-room-left-margin-width` to `0`, I don't see user names anymore. And `ement-room-right-margin-width` doesn't seem to have any effect.

**Comment by alphapapa** [comment]
> Do I experience this because I'm using Emacs with the Athena/Lucid graphical toolkit (`--with-x-toolkit=athena`)? 

No, it also happens with Gtk.  But it's good that you reported that, so now we know it isn't exclusive to Gtk.

> @alphapapa, how can I configure "Ement to not use the margin for usernames"? Which config-variable are you refering to?

Please see the message-format option.

**Comment by mekeor** [comment]
Thank you. This worked fine as a work-around:

```elisp
(custom-set-variables
 '(ement-room-message-format-spec "[%t] %S> %B%r" nil (ement-room)))
```

I keep this ticket open as the glitch/render issue is not solved.

**Comment by alphapapa** [comment]
FWIW, I don't see this bug when using Emacs 28 with Gtk, so it may be fixed in that version.

**Comment by mekeor** [comment]
I'm using 28.0.50 with Athena/Lucid/X Toolkit.

**Comment by Gavinok** [comment]
Upgraded to 28 and just like others are saying this is now gone

**Comment by alphapapa** [comment]
> I'm using 28.0.50 with Athena/Lucid/X Toolkit.

I don't know if it will help, but Emacs 28.1 is out now, so maybe try that.

**Comment by alphapapa** [comment]
AFAIK this is solved in Emacs 28.1.
